### DISCUSSION TITLE
Cost Computation for Chat App

### DISCUSSION THREAD

**Discussion by zzulueta** [microsoft/AI-Chat-App-Hack#41] [Q&A] [answered]
Do we have a cost calculator that we can use to provide assumptions on how much a chat app like this would cost end to end? The app uses a lot of components that will contribute to the cost from what I see in the resource group:
1. App Service
2. Document Intelligence
3. Azure OpenAI
4. Search Service (at Standard seems expensive as charging is present even when search is not being used)
5. Storage Account (minimal)

**Comment by zedhaque** [comment]
What’s the size of your documents in GB? and you want a working price during development correct. Production costs are hard to estimate.

**Comment by zzulueta** [comment]
No we need production costs

What would be the cost for say 15gb of company data. 
Estimated chat users per month: 5000 users
Average questions per user interaction: 10 per user
50,000 chat questions per month

We definitely can't tell customer that its hard to estimate production costs and you pay as you go

On Wednesday, January 31, 2024, Zed Haque ***@***.***> wrote:

> What’s the size of your documents in GB? and you want a working price
> during development correct. Production costs are hard to estimate.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/microsoft/AI-Chat-App-Hack/discussions/41#discussioncomment-8316730>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJ4KH2RIM2G6ZYK74XXL2BTYRIBT5AVCNFSM6AAAAABCSUW6YKVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DGMJWG4ZTA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> com>
>


-- 
Regards,
Ziggy Zulueta

Microsoft MVP Azure AI Services
8x Microsoft Certified
Microsoft Certified Azure Solutions Architect
Microsoft Certified Azure AI Engineer

**Reply by pamelafox** [reply, marked as answer]
Have you tried the cost calculator linked from https://github.com/Azure-Samples/azure-search-openai-demo/?tab=readme-ov-file#cost-estimation ?

You can estimate that each question is ~1000 tokens when using that estimator.

**Reply by zzulueta** [reply]
Thanks this would help!
